src/views/Simulation/utils/pixiUtils.js

```diff
@@ -11,92 +12,155 @@ const createGraphicsContext = (options) => {
     .stroke({ color: options.lineColor, width: 0.75 });
 };
 
-const createLinkGraphics = (links, setSelectedElement) => {
+const createLinkGraphics = (links, setSelectedElement, scale) => {
+  if (!links || !Array.isArray(links)) return [];
   return links.map((link) => {
     const graphics = new Graphics();
     graphics.alpha = 0.25;
     graphics.eventMode = 'static';
     graphics.cursor = 'pointer';
 
     const { source, target } = link;
-    graphics.moveTo(source.x + 4, source.y + 4);
+    graphics.moveTo((source.x + 4) * scale, (source.y + 4) * scale);
 
-    graphics.lineTo(target.x + 4, target.y + 4);
-    graphics.stroke({ width: 1, color: '#FFFFFF' });
+    graphics.lineTo((target.x + 4) * scale, (target.y + 4) * scale);
+    graphics.stroke({ width: 1 * scale, color: '#FFFFFF' });
     graphics.on('click', (event) => setSelectedElement(link));
     return graphics;
   });
 };
 
-export const renderElements = (sceneContainerRef, containerRef, graphRef, setSelectedElement) => {
-  if (!graphRef.current || !sceneContainerRef.current) return;
+export const renderElements = (container, graphRef, setSelectedElement, scale = 1) => {
+  if (!graphRef.current || !container) return;
+
   const { nodes, links } = graphRef.current;
+  if (!nodes || !Array.isArray(nodes) || !links || !Array.isArray(links)) return;
 
   const graphicsContexts = {
     stock: createGraphicsContext({ fillColor: '#003d00', lineColor: '#ffffff' }),
     productionResource: createGraphicsContext({ fillColor: '#3d0000', lineColor: '#ff0000' }),
     productionOperation: createGraphicsContext({ fillColor: '#00003d', lineColor: '#4444ff' }),
   };
 
-  const linkGraphics = createLinkGraphics(links, setSelectedElement);
-  linkGraphics.forEach((link) => sceneContainerRef.current.addChild(link));
+  const linkGraphics = createLinkGraphics(links, setSelectedElement, scale);
+  linkGraphics.forEach((link) => container.addChild(link));
 
   nodes.forEach((node) => {
-    const graphics = new Graphics(graphicsContexts[node.type]);
-    graphics.x = node.x;
-    graphics.y = node.y;
-    graphics.eventMode = 'static';
-    graphics.cursor = 'pointer';
-    graphics.on('click', (event) => setSelectedElement(node));
-    sceneContainerRef.current.addChild(graphics);
+    const sceneGraphics = new Graphics(graphicsContexts[node.type]);
+    sceneGraphics.x = node.x * scale;
+    sceneGraphics.y = node.y * scale;
+    sceneGraphics.scale.set(scale);
```

Comment:
I'm not sure we want to scale the whole scene like this. If we do, won't the lines be drawn thicker when we zoom in?